vulkan_hal.cpp

```diff
@@ -49,13 +50,20 @@ static VkResult AcquireImageANDROID(VkDevice, VkImage /*dev*/,
                                     int nativeFenceFd,
                                     VkSemaphore /*semaphore*/,
                                     VkFence /*fence*/) {
+  // wait for fence to signal before acquiring image
+  sync_wait(nativeFenceFd, -1);
+
   close(nativeFenceFd);
   return VK_SUCCESS;
 }
 
-static VkResult QueueSignalReleaseImageANDROID(VkQueue /*queue*/,
+static VkResult QueueSignalReleaseImageANDROID(VkQueue queue,
                                                VkImage /*image*/,
                                                int* pNativeFenceFd) {
+  // wait for any operations on queue to complete
+  // before releasing image
+  vkQueueWaitIdle(queue);
+
```

Comment:
Even if it may have performance impact (guessing) I think it is way better than what we have currently, with this frames get displayed in correct order. I don't see how this would cause deadlock, what would be the situation?